app/helpers/reference_helper.rb

```diff
@@ -0,0 +1,38 @@
+module ReferenceHelper
+
+  def after_desired_date?
```

Comment:
I'm not quiet sure about this method, but I needed to add it as repeating the method's logic all over the place seemed wrong.